packages/checkout/widgets-lib/src/widgets/sale/components/CoinsDrawerItem.tsx

```diff
@@ -1,47 +1,116 @@
 import { Heading, MenuItem } from '@biom3/react';
-import { WidgetTheme } from '@imtbl/checkout-sdk';
+import {
+  Fee,
+  FundingStepType,
+  ItemType,
+  TransactionRequirement,
+  WidgetTheme,
+} from '@imtbl/checkout-sdk';
 import {
   calculateCryptoToFiat,
   getDefaultTokenImage,
   tokenValueFormat,
 } from 'lib/utils';
 import { useTranslation } from 'react-i18next';
+import { ReactElement } from 'react';
+import { BigNumber, utils } from 'ethers';
 import { useSaleContext } from '../context/SaleContextProvider';
-import { SettlementCurrency } from '../views/balances.mock';
+import { FundingBalance } from '../types';
+
+type FeesBySymbol = Record<
+string,
+{ amount: BigNumber; formattedAmount: string; symbol: string }
+>;
+const calculateFeeAmount = (balance: FundingBalance): FeesBySymbol => {
+  let fees: Fee[] = [];
+  if (balance.type === FundingStepType.SWAP) {
+    fees = [
+      balance.fees.approvalGasFee,
+      balance.fees.swapGasFee,
+      ...balance.fees.swapFees,
```

Comment:
Just wanted to check if these swap fees are the secondary fees breakdown?

Reply:
Thanks for pointing this out, I'll need to make this a utility.

> Just wanted to check if these swap fees are the secondary fees breakdown?

No this are not for the breakdown, this is just to calc the total fee that gets presented here:
![image](https://github.com/immutable/ts-immutable-sdk/assets/2966156/2b5f5aa0-6fa8-41d3-aa5c-e48b0c5aff24)

Reply:
Ok cool, the reason I was double checking is that in the new swap fee breakdown the 1% swap fee that is quoted in the swap amount shows in the fee drawer but isn't part of the fee total (gas and approval gas)..  secondary fee will be in the from token anyway... its still in the swap quote fee data structure though.
https://files.slack.com/files-pri/T9QJC6ERM-F06V54YGERJ/image.png

Reply:
Thanks @dreamoftrees I think then in that case we are good, because the line above is meant to prompt the max fees expected via that option, but then if the client proceeds they'll see the proper breakdown during swap ui